APIAssignment/src/test/java/stepDefination/ProductBDD.java

```diff
@@ -0,0 +1,103 @@
+package stepDefination;
+
+import APIAssignments.ConfigReader;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import io.restassured.RestAssured;
+import io.restassured.path.json.JsonPath;
+import io.restassured.response.Response;
+import org.apache.log4j.PropertyConfigurator;
+import org.testng.Assert;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+import static io.restassured.RestAssured.given;
+
+public class ProductBDD {
+    private static String url;
+    private String productId;
+    private String productName;
+    private String productPrice;
+    private String productBrand;
+
+    private static Logger logger = Logger.getLogger("ProductBDD.class");
+
+    public ProductBDD() {
+        try {
+            url = ConfigReader.getUrl();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public ProductBDD(String productId, String productName, String productPrice, String productBrand) {
+        this.productId = productId;
+        this.productName = productName;
+        this.productPrice = productPrice;
+        this.productBrand = productBrand;
+    }
+
+
+    @Given("valid page number is available")
+    public void validatePageNumber() {
+        PropertyConfigurator.configure("log4j2.properties");
+    }
```

Comment:
step name and functionality does not match

Reply:
DONE!